specifyweb/frontend/js_src/lib/components/QueryBuilder/Line.tsx

```diff
@@ -231,10 +239,19 @@ export function QueryLine({
 
   const hasAny = field.filters.some(({ type }) => type === 'any');
 
+  const fieldSpec = QueryFieldSpec.fromPath(baseTableName, field.mappingPath);
```

Comment:
This might be expensive to run on every render (especially since there can be many QueryLine renders at once
Refactor to use React.useMemo